### PR TITLE
General docs polish (Solidity Utils package)

### PR DESCRIPTION
**Pull request by EndymionJkb**
# Description

Same drill, on solidity-utils. Already in pretty good shape.

## Type of change

- [ ] Bug fix 
- [ ] New feature 
- [ ] Breaking change 
- [ ] Dependency changes
- [ ] Code refactor / cleanup
- [ ] Optimization: [ ] gas / [ ] bytecode
- [X] Documentation or wording changes
- [ ] Other

## Checklist:

- [X] The diff is legible and has no extraneous changes
- [ ] Complex code has been commented, including external interfaces
- [ ] Tests have 100% code coverage
- [X] The base branch is either `main`, or there's a description of how to merge

### CODE REVIEW

File: pkg/solidity-utils/contracts/helpers/ArrayHelpers.sol [hunk 1]
```diff
@@ -4,7 +4,14 @@ pragma solidity ^0.8.24;
...
 import { IERC20 } from "@openzeppelin/contracts/token/ERC20/IERC20.sol";
 
-/// @notice Library of helper functions used to convert between fixed size and dynamic arrays of various types.
+/**
+ * @notice Library of helper functions to convert fixed-sized array types to dynamic arrays.
+ * @dev For instance, the `buildTokenConfig` functions construct a `TokenConfig` from its component parts.
+ * The first argument is a dynamic array of `IERC20`, but it's convenient to start with a fixed array of strings or
+ * addresses. `[address(dai), address(usdc)].toMemoryArray().asIERC20()` casts to the correct type.
+ *
+ * This library is intended for testing purposes; it should not be used in the main codebase.
```
Comment (joaobrunoah) [inline pkg/solidity-utils/contracts/helpers/ArrayHelpers.sol:13]: If this is only for testing, maybe we should move it to the test folder? (Maybe on another PR)

File: pkg/solidity-utils/contracts/helpers/TransientStorageHelpers.sol [hunk 2]
```diff
@@ -106,7 +114,7 @@ library TransientStorageHelpers {
         lastElementSlot.tstore(address(0));
     }
 
-    // Uint256
+    // Uint256 values
```
Comment (joaobrunoah) [inline pkg/solidity-utils/contracts/helpers/TransientStorageHelpers.sol:117]: In other files, we used the pattern:

```
/*************************
         Uint256 values
*************************/
```

Should we adopt this pattern in that files, or that pattern in this file, or are them 2 separate things?

**Review by joaobrunoah — APPROVED** [review]
Looks good!

File: pkg/solidity-utils/contracts/helpers/ArrayHelpers.sol (hunk 1 above)
Comment (EndymionJkb) [reply pkg/solidity-utils/contracts/helpers/ArrayHelpers.sol:13]: Will consider, yes. Out of scope for this PR for sure.

File: pkg/solidity-utils/contracts/helpers/TransientStorageHelpers.sol (hunk 2 above)
Comment (EndymionJkb) [reply pkg/solidity-utils/contracts/helpers/TransientStorageHelpers.sol:117]: I'd thought about that - we use the block headers everywhere else. These files were adapted from the old OZ code, which had this style. Should update to the block headers, I think.

**Pull request merged**